WordPress/src/main/java/org/wordpress/android/ui/prefs/SiteSettingsFragment.java

```diff
@@ -384,9 +382,8 @@ public boolean onPreferenceClick(Preference preference) {
         } else if (preference == mCategoryPref || preference == mFormatPref) {
             return !shouldShowListPreference((DetailListPreference) preference);
         } else if (preference == mExportSitePref) {
-            showExportContentDialog();
         } else if (preference == mDeleteSitePref) {
-            showDeleteSiteDialog();
+            requestPurchasesForDeletionCheck();
         } else {
```

Comment:
Actually I do think this is required. Some of the branches do not return, therefore the `return true` at the end.

We only absolutely return false if it didn't go into any of those conditionals.